trunner/target/armv7m4.py

```diff
@@ -0,0 +1,215 @@
+import subprocess
+import time
+from pathlib import Path
+from typing import Optional, Sequence
+
+import pexpect.fdpexpect
+import serial
+
+from trunner.config import TestContext
+from trunner.dut import Dut, SerialDut, PortError
+from trunner.harness.base import HarnessBase
+from trunner.host import Host
+from trunner.harness import (
+    IntermediateHarness,
+    TerminalHarness,
+    PloInterface,
+    PloHarness,
+    ShellHarness,
+    Rebooter,
+    RebooterHarness,
+    HarnessBuilder,
+    FlashError,
+)
+from trunner.tools import GdbInteractive, OpenocdGdbServer
+from trunner.types import AppOptions, TestOptions, TestResult
+from .base import TargetBase, find_port
+
+
+class ARMv7M4Rebooter(Rebooter):
+    # TODO add text mode
+    # NOTE: changing boot modes not needed/supported for this target
+    pass
+
+
+class STM32L4x6OpenocdGdbServerHarness(IntermediateHarness):
```

Comment:
Now output will be printed in case of gdb/plo error

Reply:
Openocd gdb server output won't be actually printed in case of failure of gdb or plo after last changes, but it's provided in case of its own error. I still think it would be good to show this output in the mentioned situations, but as we talked we could assume that if error in openocdgdbserver hadn't been caught everything was ok with that. We could eventually improve catching errors in the specific tools after merge if needed.

Reply:
These errors must be raised in the tool context. If the tools ended its work with no problems then I assume there is no reason to attach its output to the exception. As we discussed, to implement these logic we have to somehow keep the tool output in maybe global context (?) and then pass it to the plo errors. I don't have a good idea how to implement it and I don't know if this feature is that helpful.

Here Openocd gdb server output should be printed in case of failure of gdb or plo (I think). Maybe you ment gdb client?

Reply:
Yeah, I meant gdb client:
![Screenshot from 2023-02-24 15-40-14](https://user-images.githubusercontent.com/77304431/221804574-37e81a19-13ea-489e-a7a5-851f65854a09.png)
and plo:
![Screenshot from 2023-02-24 15-17-35](https://user-images.githubusercontent.com/77304431/221804698-208fd2e0-f0a4-4a6a-b9aa-27654fd38e51.png)

Ok, we can leave it like it is for now, I hope there won't be any 'hidden' errors